app/src/main/java/com/ucm/informatica/spread/Fragments/ProfileFragment.java

```diff
@@ -1,46 +1,126 @@
 package com.ucm.informatica.spread.Fragments;
 
+import android.content.Context;
+import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
 
 import com.andrognito.flashbar.Flashbar;
+import com.ucm.informatica.spread.Presenter.ProfileFragmentPresenter;
 import com.ucm.informatica.spread.R;
+import com.ucm.informatica.spread.View.ProfileFragmentView;
+import com.ucm.informatica.spread.Model.Colours;
 
-public class ProfileFragment extends Fragment {
+import static android.content.Context.MODE_PRIVATE;
+
+public class ProfileFragment extends Fragment implements ProfileFragmentView {
 
     private Button editContactsButton;
     private Button editProfileButton;
+    private Button saveProfileButton;
+    private Button[] shirtButton = new Button[Colours.values().length];
+    private Button[] pantsButton = new Button[Colours.values().length];
+    private Button editWatchwordButton;
+    private Button saveWatchwordButton;
+    private TextView nameText;
+    private TextView ageText;
+    private EditText editName;
+    private EditText editAge;
+    private TextView watchwordMessageText;
+    private TextView watchwordResponesText;
+    private EditText editWatchwordMessage;
+    private EditText editWatchwordResponse;
+
+    private Colours shirtColour = Colours.NA;
```

Comment:
shirtColour is not initialize when presenter starts so...it always has an init value of NA even though view is refreshed, you should send the color value from shared preferences...

Buttons has a stack structure so visually so at first sight, we cant not see the buttons switching below.

I suggest [this](https://github.com/patriciaTel/TFG/pull/6/commits/e328fb01888b89231465685647d2d09205df1733#r264063377)